tornado/httputil.py

```diff
@@ -62,6 +62,10 @@
     from asyncio import Future  # noqa: F401
     import unittest  # noqa: F401
 
+    StrMutableMapping = collections.abc.MutableMapping[str, str]
+else:
+    StrMutableMapping = collections.abc.MutableMapping
+
```

Comment:
Python 3.8 doesn't support type parameters on `MutableMapping`, but mypy/Pyright know how to handle them. Once Python 3.8 support is dropped, they can go right in the class definition: `HTTPHeaders(collections.abc.MutableMapping[str, str])`.